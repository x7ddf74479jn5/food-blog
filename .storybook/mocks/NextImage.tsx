/**
 * @see next/image mock
 * https://github.com/vercel/next.js/issues/18393#issuecomment-955577890
 */

import React from "react";
import * as NextImage from "next/future/image";

const OriginalNextImage = NextImage.default;

Object.defineProperty(NextImage, "default", {
  configurable: true,
  value: (/** @type {import('next/image').ImageProps} */ props) => {
    if (typeof props.src === "string") {
      return <OriginalNextImage {...props} unoptimized blurDataURL={props.src} />;
    } else {
      // don't need blurDataURL here since it is already defined on the StaticImport type
      return <OriginalNextImage {...props} unoptimized />;
    }
  },
});
