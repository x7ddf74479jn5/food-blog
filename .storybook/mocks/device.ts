export const device = {
  desktop: "PC HD",
  mobile: "iPhone12",
  tablet: "iPad Pro 12.9-in",
};
