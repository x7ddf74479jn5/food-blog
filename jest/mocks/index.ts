export { mockNextImage } from "./image";
export type { NextRouter } from "./router";
export { mockRouter, RouterContext, withMockedRouter } from "./router";
export { setupWindow } from "./window";
