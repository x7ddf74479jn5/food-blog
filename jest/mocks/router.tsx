import type { AppRouterInstance, HooksClientContext } from "mocks/next/router";
import { AppRouterContext, createMockRouterHoc } from "mocks/next/router";

export const defaultMockRouter: AppRouterInstance = {
  back: jest.fn(),
  forward: jest.fn(),
  prefetch: jest.fn(),
  push: jest.fn(),
  refresh: jest.fn(),
  replace: jest.fn(),
};

export const withMockRouter = createMockRouterHoc({
  initialMockRouter: defaultMockRouter,
});

export { AppRouterContext };
export type { AppRouterInstance, HooksClientContext };
