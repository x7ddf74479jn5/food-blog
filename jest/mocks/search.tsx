import type {
  SearchMutation,
  SearchProviderInnerProps,
  SearchState,
} from "@/components/organisms/SearchArea/SearchContext";
import { defaultSearchState, SearchProviderInner } from "@/components/organisms/SearchArea/SearchContext";

export const defaultMockSearchMutation: SearchMutation = {
  setHistory: jest.fn(),
  setSelectedCategory: jest.fn(),
  setSelectedTags: jest.fn(),
  setText: jest.fn(),
};

export const withMockSearchContext = (
  children: React.ReactElement,
  state?: Partial<SearchProviderInnerProps["state"]>
): React.ReactElement => {
  const mockSearchState: SearchState = {
    ...defaultSearchState,
    ...state,
  };

  return (
    <SearchProviderInner state={mockSearchState} mutation={defaultMockSearchMutation}>
      {children}
    </SearchProviderInner>
  );
};

export { defaultSearchState };
