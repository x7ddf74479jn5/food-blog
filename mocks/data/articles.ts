import type { TArticle } from "@/types";

import { categories } from "./categories";
import { tags } from "./tags";
import { dateCommon } from "./utils";
import { writers } from "./writers";

const articleStock = {
  id: "1",
  title: "基本の一番だしの作り方",
  writer: writers.pandashark,
  description:
    "だし昆布と鰹節（削り節）から取った出汁は、味噌汁、そば、うどん、煮物、鍋料理など、多くの和食に使用できます。",
  image: { url: "/images/5026416_s.jpg", height: 427, width: 640 },
  excerpt:
    "だし昆布と鰹節（削り節）から取った出汁は、味噌汁、そば、うどん、煮物、鍋料理など、多くの和食に使用できます。",
  body: `**分量： 800ml**

だし昆布と鰹節（削り節）から取った出汁は、味噌汁、そば、うどん、煮物、鍋料理など、多くの和食に使用できます。

## 材料
- 水：1000ml
- だし昆布：10g
- かつお削り節：40g

## 作り方
- 固く絞った布巾で、昆布の表面の汚れを軽く拭き取ります。
- 鍋に水と昆布を入れます。
- 弱火で７分ほど熱し、鍋底から小さい泡が出てきたら昆布を取り出します。
- 昆布だしを一旦沸騰させてから、削り節を入れ、弱火で１分煮出します。
- ボウルの上にザルを置き、その上にキッチンペーパーを敷いて出汁を濾します。

---

**二番だしの取り方**
1. 鍋に 500ml の水を入れて沸騰させます。
2. 鍋に一番だしのだしがらを入れ、３分煮出します。
3. 一番だしと同じ方法で濾します。

---

## 保存方法
使わない分は保存容器に移して、冷蔵庫で保存してください。冷蔵保存で 2〜3 日は持ちます。

##  ポイント
だし昆布の表面の白い粉は「うま味」成分なので、取り除かずにそのままお使いください。`,
  category: categories.rice,
  tags: [tags.preparation, tags.misoSoup, tags.udon],
  imageOption: null,
};

type TArticleCollection = {
  [key: string]: TArticle;
};

export const articles: TArticleCollection = {
  stock: { ...dateCommon, ...articleStock },
};
