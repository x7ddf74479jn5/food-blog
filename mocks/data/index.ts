export { mockArticles } from "./articles";
export { mockCategories } from "./categories";
export { mockConfig } from "./config";
export { mockPickup } from "./pickup";
export { mockTags } from "./tags";
export { dateCommon } from "./utils";
export { mockWriters } from "./writers";
