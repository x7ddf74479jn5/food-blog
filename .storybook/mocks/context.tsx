import { defaultMockRouter } from "mocks/next/router";
import { AppRouterContext } from "mocks/next/router";
import { SearchProvider } from "@/components/organisms/SearchArea/SearchContext";
import { createMockRouterHoc } from "mocks/next/router";
import React from "react";

export const withContext = (Story: React.ReactElement) => {
  return (
    <AppRouterContext.Provider value={defaultMockRouter}>
      <SearchProvider>{Story}</SearchProvider>
    </AppRouterContext.Provider>
  );
};

export const withRouterContext = createMockRouterHoc();
