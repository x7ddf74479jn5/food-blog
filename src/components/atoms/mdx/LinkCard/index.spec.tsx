import { mockArticles } from "mocks/data";
import renderer from "react-test-renderer";

import { LinkCard } from ".";

describe("components/atoms/mdx/LinkCard", () => {
  it("snapshot", () => {
    const tree = renderer.create(<LinkCard article={mockArticles.stock} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
