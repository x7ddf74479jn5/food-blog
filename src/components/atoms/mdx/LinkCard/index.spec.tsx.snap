// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/atoms/mdx/LinkCard snapshot 1`] = `
<div
  className="p-4 my-8 dark:text-white border border-gray-300 dark:border-gray-700"
>
  <article
    className="flex flex-row max-h-36 "
  >
    <a
      href="/articles/1"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <div
        className="flex-shrink-0 mt-1 mr-4 min-w-[max-content]"
      >
        <span
          style={
            Object {
              "background": "none",
              "border": 0,
              "boxSizing": "border-box",
              "display": "inline-block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "overflow": "hidden",
              "padding": 0,
              "position": "relative",
              "width": "initial",
            }
          }
        >
          <span
            style={
              Object {
                "background": "none",
                "border": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "padding": 0,
                "width": "initial",
              }
            }
          >
            <img
              alt=""
              aria-hidden={true}
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNDgiIGhlaWdodD0iNDgiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
              style={
                Object {
                  "background": "none",
                  "border": 0,
                  "display": "block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "padding": 0,
                  "width": "initial",
                }
              }
            />
          </span>
          <img
            alt="基本の一番だしの作り方"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": "cover",
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
          <noscript>
            <img
              alt="基本の一番だしの作り方"
              data-nimg="intrinsic"
              decoding="async"
              loading="lazy"
              src="/_next/image?url=%2Fimages%2F5026416_s.jpg&w=96&q=75"
              srcSet="/_next/image?url=%2Fimages%2F5026416_s.jpg&w=48&q=75 1x, /_next/image?url=%2Fimages%2F5026416_s.jpg&w=96&q=75 2x"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": "cover",
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
          </noscript>
        </span>
      </div>
    </a>
    <div
      className="flex flex-col flex-grow"
    >
      <a
        href="/articles/1"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        基本の一番だしの作り方
      </a>
      <div
        className="flex-grow"
      >
        <div
          className="flex flex-wrap gap-x-2 gap-y-1"
        >
          <a
            href="/articles/tags/preparation"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <span
              className="inline-block text-sm leading-tight text-gray-600 dark:text-gray-300"
            >
              #仕込み
            </span>
          </a>
          <a
            href="/articles/tags/miso_soup"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <span
              className="inline-block text-sm leading-tight text-gray-600 dark:text-gray-300"
            >
              #味噌汁
            </span>
          </a>
          <a
            href="/articles/tags/udon"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <span
              className="inline-block text-sm leading-tight text-gray-600 dark:text-gray-300"
            >
              #うどん
            </span>
          </a>
        </div>
      </div>
    </div>
  </article>
</div>
`;
